clip_jax/clip.py

```diff
@@ -176,14 +264,22 @@ def text_jax(text):
         return clip.encode_text(text)
 
     rng_key = jax.random.PRNGKey(42)
-    transformed = hk.transform(clip_jax)
-    jax_params = transformed.init(rng=rng_key, image=jnp.zeros((1, 3, 224, 224)), text=jnp.zeros((1, 77), dtype=jnp.int16))
-    jax_params = convert_params(state_dict, jax_params)
-
-    image_fn = hk.without_apply_rng(hk.transform(vit_jax)).apply
-    text_fn = hk.without_apply_rng(hk.transform(text_jax)).apply
-
-    return image_fn, text_fn, jax_params, _transform(clip_params["image_resolution"])
+    # todo make this optional if rn50. We only need state because of bn layers
```

Comment:
I would like to keep interface the same for vit models as to not break any code which relies on this, so it would be great if you made the state return only for rn50 (or with an optional flag if the user wants to maintain the same api between all the models) and also show an example of using the resnet model in the readme

Reply:
Yeah makes sense. I will make the state return for rn50 only